source/common/network/address_impl.h

```diff
@@ -25,8 +60,8 @@ namespace Address {
  * @param v6only disable IPv4-IPv6 mapping for IPv6 addresses?
  * @return InstanceConstSharedPtr the address.
  */
-InstanceConstSharedPtr addressFromSockAddr(const sockaddr_storage& ss, socklen_t len,
-                                           bool v6only = true);
+StatusOr<InstanceConstSharedPtr> addressFromSockAddr(const sockaddr_storage& ss, socklen_t len,
```

Comment:
There are a lot of calls to `*addressFromSockAddr` still. If those are legit (we expect to crash and not throw) then rename this addressFromSockAddrOrDie and have it just return the InstanceConstSharedPtr.

However, please be careful as the above doxygen coment indicates existing callers may expect a throw on failure, in which case calling `*addressFromSockAddr` is may crash unexpectedly.

Reply:
thanks, will fix.

Reply:
we can have addressFromSockAddrOrThrow